cli/assets/templates/rust/src/wasm4.rs

```diff
@@ -123,8 +123,9 @@ extern "C" {
 }
 
 /// Draws text using the built-in system font.
-pub fn text(text: &str, x: i32, y: i32) {
-    unsafe { extern_text(text.as_ptr(), text.len(), x, y) }
+pub fn text<T: AsRef<str>>(text: T, x: i32, y: i32) {
+    let text_ref = text.as_ref();
+    unsafe { extern_text(text_ref.as_ptr(), text_ref.len(), x, y) }
```

Comment:
Interesting, does this allow auto-casting from other string types?

Reply:
`AsRef<str>` is a [common pattern in rust api design](https://www.philipdaniels.com/blog/2019/rust-api-design/).
The gist is that you can **also** pass `String` types directly without explicitly passing a reference.
It reduces the clutter especially when using the macro `format!`

### Before

```rust
let formatted = format!("{:?}",  SMILEY);

text("string slice", 50, 50);
text(&formatted, 20, 20);
``` 


### After

```rust
text("string slice", 50, 50);
text(format!("{:?}",  SMILEY), 20, 20);
```

I've also evaluated also the possibility to add `text!` and `trace!` macros, simple wrappers for `text(format!([...]))`,  but I'm not convinced that we should provided that as default because it is not `free` in terms of bytes used.

Reply:
Good to know, thanks!